### ISSUE TITLE
Mesh regressor training data

### ISSUE THREAD

**Issue by resindraburiza** [marcbadger/avian-mesh#5]
Hello Marc Badger,
First of all I would like to thank you for releasing the code. Here I would like to ask about the mesh regressor. 

If I understand correctly, you regress the joint rotation, bone length, and global translation based on the keypoints and mask. Are you planning to release the dataset for this training? Or is it already released but I am missing it? Is it possible for us to try to train it by ourself? I would like to try training it by myself because the mesh regression network structure is simple but the predicted joint rotation, bone length, and global translation are good. 
Thank you very much.

**Comment by marcbadger** [comment]
Hi Resindra, you're correct that our regression approach takes in keypoints and a mask and outputs joint rotations, bone lengths, and translation (relative to the camera frame). The regression approach is great for initializing single-view and multi-view optimization. The optimization-only approach however, also does fairly well by itself (particularly for the multi-view setting) and is better than regression alone (see Tables 3 and S2).

We'll get back to you on the dataset/training scripts. In the meantime, it might be helpful to experiment with the optimization-only approach, either in the multiview setting [tools/multiview_demo.py](https://github.com/marcbadger/avian-mesh/blob/master/tools/multiview_demo.py), or by replacing the [regression section of tools/singleview_demo.py](https://github.com/marcbadger/avian-mesh/blob/71a884153e8970de6ee1d934949914790e40e80e/tools/singleview_demo.py#L66-L70) with some initialization lines:

`pose = torch.zeros([1, 25*3])`

`pose = batch_rodrigues(pose).view(-1, 25*9)`

`bone = torch.ones([1, 24])`

`tran = torch.zeros([1,3])`

where 25 is the number of joints including global orientation, 24 is the number of bones, and `batch_rodrigues()` is from [utils/geometry.py](https://github.com/marcbadger/avian-mesh/blob/71a884153e8970de6ee1d934949914790e40e80e/utils/geometry.py#L9) (see [OptimizeMV](https://github.com/marcbadger/avian-mesh/blob/71a884153e8970de6ee1d934949914790e40e80e/optimization/optimize_mv.py#L37-L40) or [OptimizeSV](https://github.com/marcbadger/avian-mesh/blob/71a884153e8970de6ee1d934949914790e40e80e/optimization/optimize_sv.py#L83-L85) for shapes of initialization arguments).

Looping in @yufu-wang, can we release the synthetic dataset generation and mesh_regressor training scripts that we used to obtain the models/regressor.pth weights?

**Comment by yufu-wang** [comment]
Sure, we can release the synthetic dataset generation as well. I will also give an update here when it's ready.

**Comment by resindraburiza** [comment]
Dear Marc Badger,

thank you very much for the reply.

> The optimization-only approach however, also does fairly well by itself (particularly for the multi-view setting) and is better than regression alone (see Tables 3 and S2).

I see. Thank you for pointing this out. However, as  you said, I think regression approach is great for the initialization and it helps the optimization process. 

> In the meantime, it might be helpful to experiment with the optimization-only approach

Thank you for the suggestion. Surely will try this too!

Looking forward to the release of the dataset. Thank you very much for responding to my question and request. For the mean time I will close the issue.

Best regards,
Resindra
